core/store/ChainStore/ChainStore.go

```diff
@@ -0,0 +1,996 @@
+		for index := 0; index < len(b.Transactions[i].Outputs); index++ {
+			txhash := b.Transactions[i].Hash()
+			unspents[txhash] = append(unspents[txhash], uint16(index))
+		}
+
+		// delete unspent when spent in input
+		for index := 0; index < len(b.Transactions[i].UTXOInputs); index++ {
+			txhash := b.Transactions[i].UTXOInputs[index].ReferTxID
+
+			// if get unspent by utxo
+			if _, ok := unspents[txhash]; !ok {
+				unspentValue, err_get := bd.st.Get(append(unspentPrefix, txhash.ToArray()...))
+
+				if err_get != nil {
+					return err_get
+				}
+
+				unspents[txhash], err_get = GetUint16Array(unspentValue)
+				if err_get != nil {
+					return err_get
+				}
+			}
+
+			// find Transactions[i].UTXOInputs[index].ReferTxOutputIndex and delete it
+			for k := 0; k < len(unspents[txhash]); k++ {
+				if unspents[txhash][k] == uint16(b.Transactions[i].UTXOInputs[index].ReferTxOutputIndex) {
+					unspents[txhash] = append(unspents[txhash], unspents[txhash][:k]...)
+					unspents[txhash] = append(unspents[txhash], unspents[txhash][k+1:]...)
+					break
+				}
+			}
+		}
+
+	}
+
+	// batch put the unspents
+	for txhash, value := range unspents {
+		unspentKey := bytes.NewBuffer(nil)
+		unspentKey.WriteByte(byte(IX_Unspent))
+		txhash.Serialize(unspentKey)
+
+		if len(value) == 0 {
+			bd.st.BatchDelete(unspentKey.Bytes())
+		} else {
+			unspentArray := ToByteArray(value)
+			bd.st.BatchPut(unspentKey.Bytes(), unspentArray)
+		}
+	}
+
+	// batch put quantities
+	for assetId, value := range quantities {
+		quantityKey := bytes.NewBuffer(nil)
+		quantityKey.WriteByte(byte(ST_QuantityIssued))
+		assetId.Serialize(quantityKey)
+
+		qt, err := bd.GetQuantityIssued(assetId)
+		if err != nil {
+			return err
+		}
+
+		qt = qt + value
+
+		quantityArray := bytes.NewBuffer(nil)
+		qt.Serialize(quantityArray)
+
+		bd.st.BatchPut(quantityKey.Bytes(), quantityArray.Bytes())
+		log.Debug(fmt.Sprintf("quantityKey: %x\n", quantityKey.Bytes()))
+		log.Debug(fmt.Sprintf("quantityArray: %x\n", quantityArray.Bytes()))
+	}
+
+	// save hash with height
+	bd.currentBlockHeight = b.Blockdata.Height
+
+	// generate key with SYS_CurrentHeader prefix
+	currentBlockKey := bytes.NewBuffer(nil)
+	currentBlockKey.WriteByte(byte(SYS_CurrentBlock))
+	//fmt.Printf( "SYS_CurrentHeader key: %x\n",  currentBlockKey )
+
+	currentBlock := bytes.NewBuffer(nil)
+	blockHash.Serialize(currentBlock)
+	serialization.WriteUint32(currentBlock, b.Blockdata.Height)
+
+	// BATCH PUT VALUE
+	bd.st.BatchPut(currentBlockKey.Bytes(), currentBlock.Bytes())
+
+	//bh := b.Blockdata.Hash()
+	//bd.headerIndex[bd.currentBlockHeight] = &bh
+
+	err = bd.st.BatchCommit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (bd *ChainStore) onAddHeader(header *Header) {
+	log.Debug(fmt.Sprintf("onAddHeader(), Height=%d\n", header.Blockdata.Height))
+
+	hash := header.Blockdata.Hash()
+	bd.headerIndex[header.Blockdata.Height] = hash
+	for header.Blockdata.Height-bd.storedHeaderCount >= 2000 {
```

Comment:
No Hardcode number on any place of the code, move the hard code magic number to const with some meaningful name